src/main/java/org/example/web/MainServlet.java

```diff
@@ -1,38 +1,72 @@
 package org.example.web;
 
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.ServletException;
 import jakarta.servlet.http.HttpServlet;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
-import org.example.constant.RequestArguments;
+import org.example.constant.RequestArgument;
 import org.example.model.Arguments;
-import org.example.model.Response;
+import org.example.model.JspAttribute;
+import org.example.model.JspAttributes;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.util.List;
 
 public class MainServlet extends HttpServlet {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(MainServlet.class);
     private final PathMapper pageMapService = PathMapper.getInstance();
 
     @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
         String requestURI = req.getRequestURI();
 
         LOGGER.info("We enter on path {}", requestURI);
 
+
         Arguments arguments = new Arguments();
-        arguments.getHashMap().put(RequestArguments.PATH, requestURI);
+        arguments.getHashMap().put(RequestArgument.PATH, requestURI);
 
-        Response response = pageMapService.getResponse(arguments);
+        JspAttributes response = pageMapService.getResponse(arguments);
 
         String contentType = response.getContentType();
-        resp.setContentType(contentType);
-
-        resp.getWriter().write(response.getBody());
+        response.setContentType(contentType);
 
         int code = response.getCode();
-        resp.setStatus(code);
+        response.setCode(code);
+
+        if (response.getAttributes() != null) {
+            List<JspAttribute> attributes = response.getAttributes();
+            for (JspAttribute attribute : attributes) {
+                req.setAttribute(attribute.getName(), attribute.getValue());
+            }
+        }
+
+        String jspPath = response.getResponseJspAttributes();
+
+        if (jspPath == null) {
+            try {
+                resp.getWriter().write(response.getBody());
+            } catch (IOException e) {
+                LOGGER.error("Check if response body is not null");
+                e.printStackTrace();
+                throw new RuntimeException();
+            }
+        }
+
+        RequestDispatcher requestDispatcher = req.getRequestDispatcher(jspPath);
+        try {
+            requestDispatcher.forward(req, resp);
+        } catch (ServletException | IOException e) {
+            LOGGER.error("Check if 'req' or 'resp' is not null");
```

Comment:
Strange message. Rewrite it.

(`req`, `resp` always present)

Reply:
fixed